src/AutoRest.CSharp/Generation/Writers/ClientWriter.cs

```diff
@@ -134,12 +141,122 @@ private void WriteClientFields(CodeWriter writer, Client client)
+                {
+                    writer.WriteParameterNullChecks(ctorParams);
+                    writer.Line();
+
+                    writer.Line($"{OptionsVariable} ??= new {clientOptionsName}ClientOptions();");
+                    writer.Line($"{ClientDiagnosticsField} = new {typeof(ClientDiagnostics)}({OptionsVariable});");
+                    writer.Line($"{PipelineField} = {typeof(HttpPipelineBuilder)}.Build({OptionsVariable}, new {typeof(AzureKeyCredentialPolicy)}({CredentialVariable}, \"{context.Configuration.CredentialHeaderName}\"));");
+                    writer.Append($"this.RestClient = new {client.RestClient.Type}({ClientDiagnosticsField}, {PipelineField}, ");
+                    foreach (var parameter in client.RestClient.Parameters)
+                    {
+                        if (!parameter.IsApiVersionParameter)
+                        {
+                            writer.Append($"{parameter.Name}, ");
+                        }
+                        else
+                        {
+                            writer.Append($"{OptionsVariable}.Version, ");
+                        }
+                    }
+                    writer.RemoveTrailingComma();
+                    writer.Append($");");
+                }
+                writer.Line();
+            }
+
+            if (credentialTypes.Contains("TokenCredential", StringComparer.OrdinalIgnoreCase))
+            {
+                var ctorParams = client.GetClientConstructorParameters(typeof(TokenCredential));
+                writer.WriteXmlDocumentationSummary($"Initializes a new instance of {client.Type.Name}");
+                foreach (Parameter parameter in ctorParams)
+                {
+                    writer.WriteXmlDocumentationParameter(parameter.Name, parameter.Description);
+                }
+                writer.WriteXmlDocumentationParameter(OptionsVariable, "The options for configuring the client.");
+
+                writer.Append($"public {client.Type.Name:D}(");
+                foreach (Parameter parameter in ctorParams)
+                {
+                    writer.WriteParameter(parameter);
+                }
+                writer.Append($" {clientOptionsName}ClientOptions {OptionsVariable} = null)");
+
+                using (writer.Scope())
+                {
+                    writer.WriteParameterNullChecks(ctorParams);
+                    writer.Line();
+
+                    writer.Line($"{OptionsVariable} ??= new {clientOptionsName}ClientOptions();");
+                    writer.Line($"{ClientDiagnosticsField} = new {typeof(ClientDiagnostics)}({OptionsVariable});");
+                    var scopesParam = new CodeWriterDeclaration("scopes");
+                    var credentialScopes = context.Configuration.CredentialScopes;
+                    string scope = "";
+                    if (credentialScopes.Length == 1)
+                    {
+                        scope = $"\"{credentialScopes[0]}\"";
```

Comment:
I think you can remove this branch

Reply:
Yeah, I added this if condition, so that in case of 1 scope, it would call [this](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs#L25) method

Reply:
it would still result in an array allocation so it's okay to simplify the generator code